src/components/Carousel/Carousel.spec.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2019, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+
+import Carousel from './Carousel';
+import { SLIDES } from './__fixtures__';
+
+describe('Carousel', () => {
```

Comment:
Shouldn't we also add functional tests to the Carousel?

Like: 
- clicking the next slide and expecting it to be the active one
- clicking on the last slide and expecting it to loop
- waiting for the timer to change the active slide

Reply:
yo @fernandofleury, thanks for suggestion, however all the points you mentioned are covered by the tests in useStep hook which is used inside Carousel component and is responsible for managing the state - https://github.com/sumup-oss/circuit-ui/blob/feature/add-carousel-and-step-components/src/components/Step/hooks/use-step.spec.js

Also all possible variations of internal Carousel component styles are tested separately inside components itself by passing necessarily props and matching snapshots (e.g. Slide component and different animations).

All components in Step and Carousel have 100% code coverage.

I agree that having functional/e2e test is nice addition but in current setup (react-testing-library) we will end up with just testing React onClick handlers calling the right functions. This is totally fine but it also feels like having too match hassle (e.g. adding a lot of data-selectors) for no real benefit.

WDYT?